### PR TITLE
Ac/depwarns

### PR DESCRIPTION
**Pull request by abhijithch**
Fixes deprecation warnings.

### CODE REVIEW

**Comment by tkelman** [comment]
Add the minimum `Compat` version requirement that introduced the features you're using.

File: src/BinDeps.jl [hunk 1]
```diff
@@ -361,26 +361,27 @@ module BinDeps
...
+            cmd = `make -j8`
+            if(!isempty(a.dir))
+                cmd = `$cmd -C $(a.dir)`
+            end
+            if(!isempty(a.targets))
+                cmd = `$cmd $(a.targets)`
+            end
+            @dependent_steps ( setenv(cmd, adjust_env(a.env)), )
+      end
+  end
```
Comment (tkelman) [inline src/BinDeps.jl]: be consistent with indentation. use an `else` for the windows part

File: src/BinDeps.jl [hunk 2]
```diff
@@ -361,26 +361,27 @@ module BinDeps
...
     lower(s::SynchronousStepCollection,collection) = (collection|=s)
 
     lower(s) = (c=SynchronousStepCollection();lower(s,c);c)
 
     #run(s::MakeTargets) = run(@make_steps (s,))
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+    	@static is_windows() ? (prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")) : nothing
```
Comment (tkelman) [inline src/BinDeps.jl]: use `if ... end` instead of `? : nothing`

Comment (abhijithch) [inline src/BinDeps.jl]: I went by the documentation [here](https://github.com/JuliaLang/julia/blob/master/base/osutils.jl#L56), is this

File: src/BinDeps.jl (hunk 1 above)
Comment (abhijithch) [inline src/BinDeps.jl]: Thanks, will update this.

File: src/BinDeps.jl (hunk 2 above)
Comment (tkelman) [inline src/BinDeps.jl]: It's more idiomatic to use `if end` for a single condition, the ternary requires you to go looking for a matching `:` and if what follows the `:` is just `nothing` there's no reason to use it.

File: src/BinDeps.jl [hunk 3]
```diff
@@ -298,12 +298,12 @@ module BinDeps
 
     (|)(a::BuildStep,b::BuildStep) = SynchronousStepCollection()
     function (|)(a::SynchronousStepCollection,b::SynchronousStepCollection)
-    	if(a.cwd==b.cwd)
-  		append!(a.steps,b.steps)
-    	else
-    		push!(a.steps,b)
-    	end
-    	a
+      if(a.cwd==b.cwd)
```
Comment (tkelman) [inline src/BinDeps.jl]: be consistent, with 4 spaces like the existing code

File: src/dependencies.jl [hunk 4]
```diff
@@ -468,8 +468,8 @@ function generate_steps(dep::LibraryDependency, h::Autotools,  provider_opts)
     env = Dict{String,String}()
     env["PKG_CONFIG_PATH"] = join(opts[:pkg_config_dirs],":")
     delete!(opts,:pkg_config_dirs)
-    @unix_only env["PATH"] = bindir(dep)*":"*ENV["PATH"]
-    @windows_only env["PATH"] = bindir(dep)*";"*ENV["PATH"]
+    @static is_unix() ? (env["PATH"] = bindir(dep)*":"*ENV["PATH"]) : nothing
+    @static is_windows() ? (env["PATH"] = bindir(dep)*";"*ENV["PATH"]) : nothing
```
Comment (tkelman) [inline src/dependencies.jl]: use `if ... end` here as well

File: src/dependencies.jl [hunk 5]
```diff
@@ -603,19 +606,19 @@ function check_system_handle!(ret,dep,handle)
 end
 
 # Default installation method
-if OS_NAME == :Darwin
+if Sys.KERNEL == :Darwin
```
Comment (andreasnoack) [inline src/dependencies.jl]: Use `is_apple` etc here. If you need `KERNEL` then it has to be `Compat.KERNEL`

**Comment by andreasnoack** [comment]
Superseded by #229 and #234

**Pull request closed without merging**